master/src/main/java/org/apache/celeborn/service/deploy/master/SlotsAllocator.java

```diff
@@ -144,9 +154,11 @@ static class UsableDiskInfo {
             partitionIds,
             new ArrayList<>(restriction.keySet()),
             restriction,
-            shouldReplicate);
+            shouldReplicate,
+            rackResolver);
     if (!remainPartitions.isEmpty()) {
-      roundRobin(slots, remainPartitions, new ArrayList<>(workers), null, shouldReplicate);
+      roundRobin(
```

Comment:
This call here does't guarantee slots will be allocated if rackAware is enabled. We should do as ```offerSlotsRoundRobin``` does, or we just pass null as rackResolver, and theses two places should be consistent.

Reply:
Good catch, changed